Spring-Project/src/main/java/com/epam/project/spring/controller/UserController.java

```diff
@@ -0,0 +1,49 @@
+package com.epam.project.spring.controller;
+
+import com.epam.project.spring.dto.UserDto;
+import com.epam.project.spring.model.User;
+import com.epam.project.spring.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+public class UserController {
+
+    private final UserService userService;
+
+    @ResponseStatus(HttpStatus.OK)
+    @GetMapping("/user")
+    public List<UserDto> getAllUser(){
+        return userService.listUsers();
+    }
+
+    @ResponseStatus(HttpStatus.OK)
+    @GetMapping("/user/{email}")
+    public UserDto getUser(@PathVariable String email){
+        return userService.getUser(email);
+    }
+
+    @ResponseStatus(HttpStatus.CREATED)
+    @PostMapping("/user")
+    public UserDto createUser(@RequestBody User user){
```

Comment:
Hi Zenyk. Please try to avoid using model object as **User** on the Controller level, please use DTO object instead.
For instance, we can introduce additional **ExtendedUserDto** class that will contain password field as well
or as alternative approach please send password from UI via custom Header and use **UserDto** class to send all other data. Thank you

Reply:
Thanks. Resolved